functions/mqpublish.go

```diff
@@ -41,21 +41,61 @@ func Checkin(ctx context.Context, wg *sync.WaitGroup) {
 	defer ticker.Stop()
 	ipTicker := time.NewTicker(time.Second * 15)
 	defer ipTicker.Stop()
+	checkinTicker := time.NewTicker(time.Minute * 4)
+	defer checkinTicker.Stop()
+	metricTicker := time.NewTicker(time.Minute * 5)
+	defer metricTicker.Stop()
 	for {
 		select {
 		case <-ctx.Done():
 			logger.Log(0, "checkin routine closed")
 			return
+		case <-metricTicker.C:
+			if config.CurrServer == "" {
+				continue
+			}
+			server := config.GetServer(config.CurrServer)
+			if server == nil {
+				slog.Warn("server config is nil")
+				continue
+			}
+			if Mqclient == nil || !Mqclient.IsConnectionOpen() {
+				slog.Warn("MQ client is not connected, using fallback checkin for server", config.CurrServer)
```

Comment:
if mqclient is not connected, publish metrics should be called with fallback

Reply:
added fallback, review again please.